### ISSUE TITLE
Hyperparameter search for TransE

### ISSUE THREAD

**Issue by Chen-Cai-OSU** [Accenture/AmpliGraph#124]
Hello,

I am trying different embedding methods on some synthetic data. For DistMul and ComplEx, the best hyperparameter reported [here](https://docs.ampligraph.org/en/latest/experiments.html) works pretty well. However, for TransE, I tried all the best hyperparameter for all five datasets but none of them works. 

So this means I need to perform some hyperparameter search for TransE. Would you like to provide some search range or rule of thumb? Such as which hyperparameter are important ones?

Thank you!

**Comment by sumitpai** [comment]
I would try
```
{
   'epochs': [500],
   'eta':[5, 15, 25]
   'batches_count': [100], # increase this if you get memory error
   'k': [450, 350, 250],
   'loss': ['self_adversarial'], #in the next release you would have multiclass loss which works better for most cases
   'loss_params': {
      margin: [0.5, 1, 3],
      alpha: [0.5, 1]
   },
   'regularizer': [None, 'LP'],
   'regularizer_params': {
      'p':[2,3],
      'lambda':[1e-3]
   },
   optimizer: ['adam'],
   optimizer_params: {
      'lr': [5e-4, 5e-5]
   },
   'seed': 0
}


# Use early stopping:
{
   'x_valid' : x_valid,
   'x_filter': x_filter,
   'criteria': 'mrr', # or hits10
   'corrupt_side': 's+o',
   'burn_in': 50,
   'check_interval': 50,
   'stop_interval': 3
}
   

```
This would result in more than 100 runs of hyper param search runs. This is the usual grid search params I would try. In the next release we also have random search. 

You can also quickly try a single run using the max values of the params from the above grid.

It could also be that the dataset contains a lot of inverse relations, etc which TransE cant handle well and that's why you are not getting expected results.

**Comment by sumitpai** [comment]
Can you lower the learning rate by a factor of 10 or 100 and add heavy regularization (LP with p=3 and lambda=1e-2) and see what happens?

**Comment by tabacof** [comment]
Another option is trying a different loss, like `'nll'` instead of self_adversarial.

**Comment by sumitpai** [comment]
Are you using self_adversarial loss implementation of master branch? There was a numerical instability with it due to log (sigmoid). This has been fixed on develop branch. I guess that would solve it.
